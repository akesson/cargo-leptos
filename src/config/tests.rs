use camino::Utf8PathBuf;

use super::Config;

fn opts(project: Option<&str>) -> crate::Opts {
    crate::Opts {
        release: false,
        project: project.map(|s| s.to_string()),
        verbose: 0,
        bin_features: Vec::new(),
        lib_features: Vec::new(),
    }
}

#[test]
fn test_project() {
    let cli = opts(None);

    let conf = Config::load(cli, &Utf8PathBuf::from("examples/project/Cargo.toml"), true).unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace() {
    let cli = opts(None);

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project1() {
    let cli = opts(Some("project1"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}

#[test]
fn test_workspace_project2() {
    let cli = opts(Some("project2"));

    let conf = Config::load(
        cli,
        &Utf8PathBuf::from("examples/workspace/Cargo.toml"),
        true,
    )
    .unwrap();

    insta::assert_debug_snapshot!(conf);
}
