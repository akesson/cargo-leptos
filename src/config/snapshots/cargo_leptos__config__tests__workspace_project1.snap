---
source: src/config/tests.rs
expression: conf
---
Config {
    projects: [
        Project {
            name: "project1",
            config: ProjectConfig {
                output_name: "front_package",
                site_addr: 127.0.0.1:3000,
                site_root: "target/site/project1",
                site_pkg_dir: "pkg",
                style_file: Some(
                    "project1/css/main.scss",
                ),
                assets_dir: Some(
                    "project1/assets",
                ),
                reload_port: 3001,
                end2end_cmd: None,
                end2end_dir: None,
                browserquery: "defaults",
                bin_target: "server-package",
                lib_features: [],
                lib_default_features: false,
                bin_features: [],
                bin_default_features: false,
                config_dir: "",
            },
            front_package: "front-package ..",
            front_profile: "dev",
            server_package: "server-package ..",
            server_target: "server-package (bin) ..",
            server_profile: "dev",
            watch: true,
            site: Site {
                file_reg: {},
                ext_file_reg: {},
            },
            paths: ProjectPaths {
                lib_crate_name: "front_package",
                site_root: "target/site/project1",
                site_pkg_dir: "target/site/project1/pkg",
                front_dir: "project1/front",
                server_dir: "project1/server",
                target_dir: "target",
                wasm_file: SourcedSiteFile {
                    source: "target/front/wasm32-unknown-unknown/debug/front_package.wasm",
                    dest: "target/site/project1/pkg/front_package.wasm",
                    site: "pkg/front_package.wasm",
                },
                js_file: SiteFile {
                    dest: "target/site/project1/pkg/front_package.js",
                    site: "pkg/front_package.js",
                },
                style_file: Some(
                    SourcedSiteFile {
                        source: "project1/css/main.scss",
                        dest: "target/site/project1/pkg/front_package.css",
                        site: "pkg/front_package.css",
                    },
                ),
                assets_dir: Some(
                    "project1/assets",
                ),
                cargo_bin_file: "target/server/debug/server-package",
                ..
            },
            ..
        },
    ],
    cli: Opts {
        release: false,
        project: Some(
            "project1",
        ),
        lib_features: [],
        bin_features: [],
        verbose: 0,
    },
    watch: true,
}
